### DISCUSSION TITLE
':kspKotlin' always failed

### DISCUSSION THREAD

**Discussion by hantsy** [Kotlin/dataframe#349] [Q&A] [answered]
Upgraded to DataFrame 0.10.0, when I ran `./gradlew formatKotlin`, the `kspKotlin` task always failed with the following build info.

```
BUILD FAILED in 11s
2 actionable tasks: 1 executed, 1 up-to-date
\nlintKotlin found problems, running formatKotlin; commit the result and re-push
> Task :generateDataFrames UP-TO-DATE
> Task :generateJooq UP-TO-DATE
> Task :kspKotlin FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':kspKotlin'.
> Number of loaded files in snapshots differs. Reported changed files: [C:\Users\hantsy\.gradle\caches\transforms-3\74c9b2bcf236f6238766a28a7d9bc976\transformed\output.bin, C:\Users\hantsy\.gradle\caches\transforms-3\cfe6f6d85df42eba923bf780c028a34f\transformed\output.bin, C:\Users\hantsy\.gradle\caches\transforms-3\1049666e007879ee7ccd25156eed28c8\transformed\output.bin, C:\Users\hantsy\.gradle\caches\transforms-3\3d24ae0354a443fae43af42e85c7fbc1\transformed\output.bin, C:\Users\hantsy\.gradle\caches\transforms-3\ff358e6998dde747f892226e7d9b7a56\transformed\output.bin, C:\Users\hantsy\.gradle\caches\transforms-3\d265776e800bc9e309b3571b152f3fde\transformed\output.bin, C:\Users\hantsy\.gradle\caches\transforms-3\8d3ed665d238d987774d041b14f810c0\transformed\output.bin, C:\Users\hantsy\.gradle\caches\transforms-3\bc8b2053675349515d70f05362a8f029\transformed\output.bin, C:\Users\hantsy\.gradle\caches\transforms-3\32010ccc016442f889fbe0e1b18f7e5b\transformed\output.bin,....

```

Unless I use a clean task before running formatKotlin, `./gradlew clean formatKotlin` will pass, but it always format the generated codes by KSP plugin, as described in #348

**Comment by Jolanrensen** [comment]
That's odd, I cannot reproduce it on my machine. Might have to do something with the upgrade to Gradle 8. I had to change some `dependsOn` calls in tasks, because some implicit dependencies existed. Maybe that has something to do with it.

**Reply by hantsy** [reply]
Yes, we have used Gradle 8.0 for a while, so have to setup the dependsOn, else the build will fail.

**Comment by Jolanrensen** [comment]
https://github.com/google/ksp/issues/1280 seems to appear here too

**Comment by koperagen** [comment]
0.10.0-dev-1532 uses KSP 1.0.9, should not be affected by this bug. I can suggest sticking to an older version for now, until we publish a new version with KSP hotfix release

**Comment by hantsy** [comment]
https://github.com/google/ksp/releases/tag/1.8.20-1.0.11 Hope dataframe will release a hotfix version, I still can not exclude the KSP generated codes and can not  avoid `Execution failed for task ':lintKotlinGeneratedByKspKotlin'`

**Comment by hantsy** [comment, marked as answer]
Temporarily add the lastest KSP plugin explicitly in our project to resolve this issue. 

```kotlin 
id("com.google.devtools.ksp") version "1.8.20-1.0.11"
```

**Reply by hantsy** [reply]
But this plugin is used by DataFrame directly, not used in our project.

I hope DataFrame will release a hotfix version to resolve this issue and the format issue of the KSP plugin generated codes #348 permanently.
